presto-main/src/main/java/com/facebook/presto/metadata/ConnectorMetadataUpdateHandleSerdeManager.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.connector.ConnectorMetadataUpdateHandleSerde;
+import com.facebook.presto.spi.connector.ConnectorMetadataUpdateHandleSerdeProvider;
+import com.google.inject.Inject;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorMetadataUpdateHandleSerdeManager
+{
+    private final Map<ConnectorId, ConnectorMetadataUpdateHandleSerdeProvider> metadataUpdateHandleSerdeProviderMap = new ConcurrentHashMap<>();
+
+    @Inject
+    public ConnectorMetadataUpdateHandleSerdeManager() {}
+
+    public void addMetadataUpdateHandleSerdeProvider(ConnectorId connectorId, ConnectorMetadataUpdateHandleSerdeProvider metadataUpdateHandleSerdeProvider)
+    {
+        requireNonNull(connectorId, "connectorId is null");
+        requireNonNull(metadataUpdateHandleSerdeProvider, "metadataUpdateHandleSerdeProvider is null");
+        checkArgument(metadataUpdateHandleSerdeProviderMap.putIfAbsent(connectorId, metadataUpdateHandleSerdeProvider) == null,
```

Comment:
leave the first line empty and move the first param to its own line

Reply:
done